src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row.tsx

```diff
@@ -98,54 +95,62 @@ export const GridRow = React.memo(({ rowIndex }: GridRowProps) => {
        * reasons (screen readers and focus management).
        */
       const gridLayoutSubscription = gridLayoutStateManager.gridLayout$.subscribe((gridLayout) => {
+        if (!gridLayout[rowIndex]) return;
         const newPanelIdsInOrder = getKeysInOrder(gridLayout[rowIndex].panels);
         if (panelIdsInOrder.join() !== newPanelIdsInOrder.join()) {
           setPanelIdsInOrder(newPanelIdsInOrder);
         }
       });
 
-      const columnCountSubscription = gridLayoutStateManager.runtimeSettings$
-        .pipe(
-          map(({ columnCount }) => columnCount),
-          distinctUntilChanged()
-        )
-        .subscribe((columnCount) => {
-          const rowRef = gridLayoutStateManager.rowRefs.current[rowIndex];
-          if (!rowRef) return;
-          rowRef.style.setProperty('--kbnGridRowColumnCount', `${columnCount}`);
-        });
-
       return () => {
         interactionStyleSubscription.unsubscribe();
         gridLayoutSubscription.unsubscribe();
         rowStateSubscription.unsubscribe();
-        columnCountSubscription.unsubscribe();
       };
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [rowIndex]
   );
 
+  const toggleIsCollapsed = useCallback(() => {
+    const newLayout = cloneDeep(gridLayoutStateManager.gridLayout$.value);
+    newLayout[rowIndex].isCollapsed = !newLayout[rowIndex].isCollapsed;
+    gridLayoutStateManager.gridLayout$.next(newLayout);
+  }, [rowIndex, gridLayoutStateManager.gridLayout$]);
+
+  useEffect(() => {
+    /**
+     * Set `aria-expanded` without passing the expanded state as a prop to `GridRowHeader` in order
+     * to prevent `GridRowHeader` from rerendering when this state changes
+     */
+    if (!collapseButtonRef.current) return;
```

Comment:
this is an interesting pattern to avoid re-rendering of the component, but does it make sense to put it here? Maybe yes, if in the future we're planning adding things like controls and other heavy components in the header. So just wondering, since now it's very small and rerendering costs nothing.

Reply:
It's ok to leave it as is!